src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/FlatFWTest.java

```diff
@@ -133,165 +212,171 @@ public void shouldFailToSetString1BeforeFixed1() throws Exception
-            .fixed1(101)
-            .build();
+        setAllRequiredBuilderFields(flatRW.wrap(buffer, 10, 14), INDEX_STRING1)
+                .string1("1234");
     }
 
     @Test
-    public void shouldFailToResetString1() throws Exception
+    public void shouldSetString1ValuesToNull() throws Exception
     {
-        expectedException.expect(IllegalStateException.class);
-        expectedException.expectMessage("string1");
-        flatRW.wrap(buffer, 0, 100)
-            .fixed1(10)
-            .fixed2(111)
-            .string1("value1")
-            .string1("another value")
-            .build();
+        flatRW.wrap(buffer, 0, buffer.capacity())
+                .fixed1(10)
+                .fixed2(20)
+                .string1((String) null);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void shouldFailToSetFixed3WithInsufficientSpace()
+    {
+        setAllRequiredBuilderFields(flatRW.wrap(buffer, 10, 15), INDEX_FIXED3)
+                .fixed3(30);
     }
 
     @Test
-    public void shouldFailToBuildWhenFixed1NotSet() throws Exception
+    public void shouldFailToSetString2BeforeFixed1() throws Exception
     {
         expectedException.expect(IllegalStateException.class);
         expectedException.expectMessage("fixed1");
         flatRW.wrap(buffer, 0, 100)
-            .build();
+                .string2("value1");
     }
 
     @Test
-    public void shouldFailToBuildWhenString1NotSet() throws Exception
+    public void shouldFailToSetString2BeforeString1() throws Exception
     {
         expectedException.expect(IllegalStateException.class);
         expectedException.expectMessage("string1");
-        flatRW.wrap(buffer, 0, 100)
-            .fixed1(10)
-            .build();
+        setAllRequiredBuilderFields(flatRW.wrap(buffer, 0, 100), INDEX_STRING1)
+                .string2("value1");
     }
 
     @Test
     public void shouldFailToBuildWhenString2NotSet() throws Exception
     {
         expectedException.expect(IllegalStateException.class);
         expectedException.expectMessage("string2");
-        flatRW.wrap(buffer, 0, 100)
-            .fixed1(10)
-            .fixed2(111)
-            .string1("value1")
-            .fixed3(33)
-            .build();
+        setAllRequiredBuilderFields(flatRW.wrap(buffer, 0, 100), INDEX_STRING2)
+                .build();
     }
 
-    @Test
-    public void shouldSetAllValues() throws Exception
+    private void checkAllBuilderValuesBySetterType(SetterType setterType)
     {
-        flatRW.wrap(buffer, 0, buffer.capacity())
-                .fixed1(10)
-                .fixed2(20)
-                .string1("value1")
+        final int offset = 11;
+        int expectedLimit = setAllBufferValues(expected, offset);
+
+        Builder builder = flatRW.wrap(buffer, offset, expectedLimit);
+
+        builder.fixed1(10);
+        builder.fixed2(20);
+
+        switch (setterType)
+        {
+            case STRING: builder.string1("value1")
```

Comment:
Just test the setXxx(String) variant. We'll cover the others in the generated StringFWTest